File: MyProject/C++/NoArvore.cpp
#include "NoArvore.hpp"

NoArvore::NoArvore(char caractere, int frequencia)
    : No(frequencia), caractere(caractere), esquerda(nullptr), direita(nullptr) {}

NoArvore::NoArvore(int frequencia, NoArvore* esquerda, NoArvore* direita)
    : No(frequencia), caractere('\0'), esquerda(esquerda), direita(direita) {}

NoArvore::~NoArvore() {
    delete esquerda;
    delete direita;
}

char NoArvore::getCaractere() const { return caractere; }
void NoArvore::setCaractere(char c) { this->caractere = c; }

NoArvore* NoArvore::getEsquerda() const { return esquerda; }
void NoArvore::setEsquerda(NoArvore* esq) { this->esquerda = esq; }

NoArvore* NoArvore::getDireita() const { return direita; }
void NoArvore::setDireita(NoArvore* dir) { this->direita = dir; }

bool NoArvore::ehFolha() const {
    return esquerda == nullptr && direita == nullptr;
}

File: MyProject/NoArvore.java
public class NoArvore extends No {
    
    private char caractere;
    private NoArvore esquerda;
    private NoArvore direita;

    public NoArvore(char caractere, int frequencia) {
        super(frequencia);
        this.caractere = caractere;
        this.esquerda = null;
        this.direita = null;
    }

    public NoArvore(int frequencia, NoArvore esquerda, NoArvore direita) {
        super(frequencia);
        this.caractere = '\0';
        this.esquerda = esquerda;
        this.direita = direita;
    }

    public char getCaractere() {
        return caractere;
    }

    public void setCaractere(char caractere) {
        this.caractere = caractere;
    }

    public NoArvore getEsquerda() {
        return esquerda;
    }

    public void setEsquerda(NoArvore esquerda) {
        this.esquerda = esquerda;
    }

    public NoArvore getDireita() {
        return direita;
    }

    public void setDireita(NoArvore direita) {
        this.direita = direita;
    }

    public boolean ehFolha() {
        return esquerda == null && direita == null;
    }
}